tsdb/tsm1/engine_delete_prefix.go

```diff
@@ -77,24 +77,30 @@ func (e *Engine) DeleteBucketRange(name []byte, min, max int64) error {
 
 	// ApplySerialEntryFn cannot return an error in this invocation.
 	_ = e.Cache.ApplyEntryFn(func(k []byte, _ *entry) error {
-		if bytes.HasPrefix(k, name) {
-			if deleteKeys == nil {
-				deleteKeys = make([][]byte, 0, 10000)
-			}
-			deleteKeys = append(deleteKeys, k)
+		if !bytes.HasPrefix(k, name) {
+			return nil
+		}
+		if pred != nil && !pred.Matches(k) {
```

Comment:
That doesn't work because `pred` is an interface. If it's actually a nil interface, there's no receiver and it will panic, even if every implementation contains the check.